### ISSUE TITLE
`attr_reader` vs. `attr_accessor`

### ISSUE THREAD

**Issue by scepticalcat** [KarthickNarayana/toy_robot#9]
https://github.com/KarthickNarayana/toy_robot/blob/3d570a92d5c8b027e2456ec4f7bce47223719ecf/lib/toy_robot/robot.rb#L20

You've used `attr_reader` to define implicit getters, but as you're also setting the values for `@h_position` etc, then it would make sense to use `attr_accessor` to likewise implicitly define setters, avoiding direct manipulation of the `@` vars.

**Comment by KarthickNarayana** [comment]
Yes Direct manipulation of @ vars is not a good practice. Will be using accessors and readers to do it.
